stripe/_search_result_object.py

```diff
@@ -54,6 +55,23 @@ def search(self, **params: Mapping[str, Any]) -> Self:
             ),
         )
 
+    async def _search_async(self, **params: Mapping[str, Any]) -> Self:
+        url = self.get("url")
+        if not isinstance(url, str):
+            raise ValueError(
+                'Cannot call .list on a list object without a string "url" property'
+            )
```

Comment:
Could we move this into a `_get_url_for_search`?

Reply:
Done in [_get_url_for_search and _maybe_empty_result](https://github.com/stripe/stripe-python/pull/1219/commits/d38887b367d0b0cd8c63aef60debd6c86840e4d1)